app/src/main/java/com/compose/quotes_jetpack/MainActivity2.kt

```diff
@@ -105,12 +107,90 @@ class MainActivity2 : ComponentActivity() {
                                 fontWeight = FontWeight.Bold,
                                 fontSize = 15.sp
                             )
+
+                            Image(
+                                painter = painterResource(R.drawable.baseline_cancel_24),
+                                contentDescription = null,
+                                colorFilter = ColorFilter.tint(Color.Black),
+                                modifier = Modifier.padding(start = 280.dp)
+                                    .clickable {
+                                        opendialog.value = true
+                                        val data1 = it.favauthor
+                                        val data2 = it.favquote
+                                        CoroutineScope(Dispatchers.IO).launch {
+                                            try {
+                                                val querySnapshot = reff.whereEqualTo("favauthor", data1)
+                                                    .whereEqualTo("favquote", data2).get().await()
+                                                if (querySnapshot.documents.isNotEmpty()) {
+                                                    for (i in querySnapshot.documents) {
+                                                        docid = i.id
+                                                    }
+
+
+                                                }
+
+                                            } catch (e: Exception) {
+                                                withContext(Dispatchers.Main) {
+                                                    Toast.makeText(context, e.message, Toast.LENGTH_LONG).show()
+                                                }
+                                            }
+
+                                        }
+
+
+                                    }
+                            )
                         }
                     }
                 }
             }
         }
-    }    }
+        if (opendialog.value) {
+            AlertDialog(
+                onDismissRequest = { opendialog.value = false },
+                title = {
+                    Text(text = "Quotes App", fontStyle = FontStyle.Italic, fontWeight = FontWeight.Bold)
+                },
+                text = {
+                    Text(
+                        text = "Do you want to remove the quote from favourites",
+                        fontStyle = FontStyle.Italic,
+                        fontWeight = FontWeight.Bold
+                    )
+                },
+                confirmButton = {
+                    Button(onClick = {
+                        reff.addSnapshotListener { querySnapshot, FirebaseFireStoreException ->
```

Comment:
So here the confirm button when clicked holds a snapshot listener which is used for getting the realtime updates like as the user deletes the quote then it would be removed from the list that is been seen by the user currently.
Then just call the delete functionality and once the document is successfully deleted we would be dismissing the dialog box.